### DISCUSSION TITLE
ClickSmart 30036 double socket

### DISCUSSION THREAD

**Discussion by kjoliver** [Koenkk/zigbee2mqtt#10831] [Devices]
This product is recognised as a tuya-ts011f_2_gang_wall< but the available Functionality of local locking is unavailable along with a schedule function ?

Is this my mistake ?

Kevin

**Comment by kjoliver** [comment]
I forgot to mention I'm using 

"version": "1.21.2-dev"
 Kevin

**Comment by simonaldrich** [comment]
Did you get anywhere with the local locking? I've also got one of these ClickSmart+ double sockets and the child lock function would be super useful.

**Reply by kjoliver** [reply]
Hi , no not yet still trying to get the info

**Comment by simonaldrich** [comment]
I've worked it out :slightly_smiling_face: There's a boolean attribute `0x8000` of cluster `0x0006`. Setting it `true` engages the child-lock (it applies to both outlets btw) and setting it `false` disengages the child-lock.

I was looking for how to use this from Tasmota (you can read the thread [here](https://github.com/arendst/Tasmota/discussions/14727)) but hopefully this information will be helpful to you.

All the best, Simon :slightly_smiling_face:

**Reply by jampez77** [reply]
Hey @simonaldrich You don't happen to know how I would expose / use the child lock in Z2M by any chance?

**Reply by jampez77** [reply]
I've tried setting the attribute / clusters you mentioned but i've had no joy at all. I must be doing something wrong. haha

**Reply by simonaldrich** [reply]
Sorry, no idea I'm afraid. I'm a ZHA (via Home Assistant) user. Best of luck though.

**Reply by jampez77** [reply]
Thanks man. Does the lock feature appear in ZHA?

**Reply by jampez77** [reply]
FYI, the latest version of Z2M now supports the child lock feature on these sockets.

**Reply by simonaldrich** [reply]
Thanks, FYI I've also raised a [PR](https://github.com/zigpy/zha-device-handlers/pull/2085) to support the child-lock feature in ZHA. :slightly_smiling_face:
